source/Character.cpp

```diff
@@ -1,49 +1,148 @@
 #include"Character.hpp"
-#include"Error.hpp"
-#include"Item.hpp"
-#include"Game.hpp"
 
 #include<memory>
+#include<utility>
 
 namespace dungeon {
 
-Character::Character(): curX(0), curY(0) {
-	critRateBase = 5, critDamageBase = 50;
-	inventory.addItem(std::make_unique<WreckingBar>());
-}
-
-void Character::initialize() {
-	switch (type) {
-	case Class::saber:
-		hpBase = 2222, atkBase = 1734;
-		break;
-	case Class::archer:
-		hpBase = 2134, atkBase = 1742;
-		break;
-	case Class::lancer:
-		hpBase = 2288, atkBase = 1699;
-		break;
-	// case Class::rider:
-	// 	hpBase = 2090, atkBase = 1665;
-	// 	break;
-	// case Class::caster:
-	// 	hpBase = 2112, atkBase = 1629;
-	// 	break;
-	// case Class::assassin:
-	// 	hpBase = 1862, atkBase = 1817;
-	// 	break;
-	case Class::berserker:
-		hpBase = 2079, atkBase = 1717;
-		break;
+void Character::move(DirectionType direction) {
+	switch (direction) {
+	case DirectionType::UP:
+		curX -= 1; break;
+	case DirectionType::DOWN:
+		curX += 1; break;
+	case DirectionType::LEFT:
+		curY -= 1; break;
+	case DirectionType::RIGHT:
+		curY += 1; break;
 	default:
-		throw TypeError();
+		std::unreachable();
 	}
-	curHp = hp();
 }
 
 template<>
-CharNA<Class::saber>::CharNA() {
+CharacterClass<ClassType::SABER>::CharacterClass(const std::string& name) {
+	type = ClassType::SABER;
+	this->name = name;
 
+	attr.baseHp = 2222;
```

Comment:
Still, better to create the constructor in base class (Character) to reuse codes as all of the constructor of the child classes look the same except for the parameters.

Reply:
So I don't need the derived classes?